src/workload.rs

```diff
@@ -100,18 +100,18 @@ pub struct GatewayAddress {
 pub mod gatewayaddress {
     use super::{NamespacedHostname, NetworkAddress};
     #[derive(Debug, Hash, Eq, PartialEq, Clone, serde::Serialize, serde::Deserialize)]
-    #[serde(tag = "address", content = "content")]
+    #[serde(untagged)]
     pub enum Destination {
-        Hostname(NamespacedHostname),
         Address(NetworkAddress),
+        Hostname(NamespacedHostname),
```

Comment:
at the very least I think we need a test here to ensure we can serialize / deserialize both types here

the reason I added tags in the first place was because in earlier versions of the code (when hostname was just a string) it would always serialize as hostname even if an address was provided, so i opted to be explicit about the type to avoid bugs

Reply:
I think this test coverage can be provided by the linked admin.rs config dump test (serialize) and the local_client test that deserializes from yaml